gr_modtool bind not working on 3.10

### What happened?

getting pygccxml error when using `gr_modtool bind` with an OOT created with GR 3.10

### System Information

OS: Ubuntu 20.04
GR Installation Method: source


### GNU Radio Version

3.10-git (master)

### Specific Version

3.10.0.0-rc1

### Steps to Reproduce the Problem

gr_modtool newmod asdf
cd gr-asdf
gr_modtool add -t sync -l cpp foo
gr_modtool bind foo

### Relevant log output

```shell
(('/usr/include/c++/9/type_traits', 599), ('::', 'std', 'type'))
```

Interesting, pygccxml wasn't installed. When I installed it, castxml was missing. When I installed that I got
```
$ gr_modtool bind foo

GNU Radio module name identified: asdf
INFO Parsing source file "/home/jwl/grmaster/gr-asdf/include/gnuradio/asdf/foo.h" ... 
In file included from /home/jwl/grmaster/gr-asdf/include/gnuradio/asdf/foo.h:12:
In file included from /home/jwl/grmaster/include/gnuradio/sync_block.h:15:
In file included from /home/jwl/grmaster/include/gnuradio/block.h:17:
In file included from /home/jwl/grmaster/include/gnuradio/basic_block.h:15:
In file included from /home/jwl/grmaster/include/gnuradio/io_signature.h:17:
In file included from /home/jwl/grmaster/include/gnuradio/buffer_double_mapped.h:15:
In file included from /home/jwl/grmaster/include/gnuradio/buffer.h:15:
In file included from /home/jwl/grmaster/include/gnuradio/custom_lock.h:15:
In file included from /home/jwl/grmaster/include/gnuradio/logger.h:27:
In file included from /usr/include/spdlog/common.h:36:
/usr/include/spdlog/fmt/fmt.h:22:14: fatal error: 'spdlog/fmt/bundled/core.h' file not found
#    include <spdlog/fmt/bundled/core.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
Error occurred while running CASTXML:  status:1
```

--- a/gnuradio-runtime/include/gnuradio/logger.h
+++ b/gnuradio-runtime/include/gnuradio/logger.h
@@ -17,6 +17,17 @@
  * \brief GNU Radio logging wrapper
  *
  */
+#ifdef DISABLE_LOGGER_H
+// pygccxml as of v2.2.1 has a difficult time parsing headers that
+// include spdlog or format
+// Since it only needs the top level header info, this is a hack to not
+// transitively include anything logger related when parsing the
+// headers
+#include <memory>
+namespace gr {
+using logger_ptr = std::shared_ptr<void>;
+}
+#else
 
 // Since this file is included in *all* gr::blocks, please make sure this list of includes
 // keeps as short as possible; if anything is needed only by the implementation in
@@ -276,5 +287,7 @@ struct fmt::formatter<boost::format> : formatter<string_view> {
         return formatter<string_view>::format(bfmt.str(), ctx);
     }
 };
+
+#endif
 
 #endif /* INCLUDED_GR_LOGGER_H */

--- a/gnuradio-runtime/python/gnuradio/gr/bindings/logger_python.cc
+++ b/gnuradio-runtime/python/gnuradio/gr/bindings/logger_python.cc
@@ -15,7 +15,7 @@
 /* BINDTOOL_GEN_AUTOMATIC(0)                                                       */
 /* BINDTOOL_USE_PYGCCXML(0)                                                        */
 /* BINDTOOL_HEADER_FILE(logger.h)                                        */
-/* BINDTOOL_HEADER_FILE_HASH(7f3cbb9463e52829b27c58d9dd41b422)                     */
+/* BINDTOOL_HEADER_FILE_HASH(1b018cc2df46366b8b83a4fe08db2806)                     */
 /***********************************************************************************/
 
 #include <pybind11/complex.h>

--- a/gr-utils/blocktool/core/parseheader_generic.py
+++ b/gr-utils/blocktool/core/parseheader_generic.py
@@ -53,7 +53,7 @@ class GenericHeaderParser(BlockTool):
         BlockTool.__init__(self, **kwargs)
         self.parsed_data = {}
         self.addcomments = blocktool_comments
-        self.define_symbols = ('BOOST_ATOMIC_DETAIL_EXTRA_BACKEND_GENERIC',)
+        self.define_symbols = ('BOOST_ATOMIC_DETAIL_EXTRA_BACKEND_GENERIC', 'DISABLE_LOGGER_H')
         if(define_symbols):
             self.define_symbols += define_symbols
         self.include_paths = None

